Parsers/ASimNetworkSession/Parsers/vimNetworkSessionForcePointFirewall.yaml

```diff
@@ -0,0 +1,416 @@
+            | where ((array_length(dvcaction) == 0) or DvcAction has_any (dvcaction))
+            | where ((eventresult == "*") or (EventResult == eventresult))
+            | lookup ApplicationProtocolLookup on ApplicationProtocol
+            | extend 
+                    EventCount = toint(1),
+                    EventType = "NetworkSession",
+                    EventSchema = "NetworkSession",
+                    EventSchemaVersion = "0.2.6",
+                    EventVendor = "Forcepoint",
+                    EventProduct = "Firewall"
+            | parse AdditionalExtensions with * "requestURL=" requestURL            
+            | project-rename
+                    EventOriginalType = DeviceEventClassID,
+                    DstPortNumber = DestinationPort,
+                    DstIpAddr = DestinationIP,
+                    SrcPortNumber = SourcePort,
+                    SrcIpAddr = SourceIP,
+                    DstNatIpAddr = DestinationTranslatedAddress,
+                    DstNatPortNumber = DestinationTranslatedPort,
+                    SrcNatIpAddr = SourceTranslatedAddress,
+                    SrcNatPortNumber = SourceTranslatedPort,
+                    EventProductVersion = DeviceVersion,
+                    EventMessage = Message,
+                    DvcOriginalAction = DeviceAction,
+                    SrcBytes = SentBytes,
+                    DstBytes = ReceivedBytes,
+                    EventOriginalSubType = DeviceFacility,
+                    DvcId = DeviceExternalID,
+                    DvcInboundInterface = DeviceInboundInterface,
+                    DvcOutboundInterface = DeviceOutboundInterface,
+                    DvcIpAddr = DeviceAddress,
+                    EventOriginalSeverity = LogSeverity,
+                    ThreatId = DeviceCustomString3,
+                    ThreatCategory = DeviceEventCategory
+            | invoke _ASIM_ResolveDvcFQDN('Computer')
+            | extend
+                    EventStartTime = todatetime(ReceiptTime),
+                    EventEndTime = todatetime(ReceiptTime),
+                    ipv6_parts = extract_all (@'^\[(.+)\](?:\:(\d+))?$',requestURL)[0],
+                    ipv4_parts = extract_all (@'^(\d+\.\d+\.\d+\.\d+)(?:\:(\d+))?$',requestURL)[0],
+                    host_parts = extract_all (@'^([^\\\d:]+)(?:\:(\d+))?$',requestURL)[0]
+            | extend 
+                     NetworkRuleName = case(isnotempty(DeviceCustomString2), strcat(DeviceCustomString1,',',DeviceCustomString2),
+                                            DeviceCustomString1),
+                     DstDomainPart = tostring(host_parts[0]),
+                     DstIpAddr = coalesce(DstIpAddr, ipv4_parts[0],ipv6_parts[0]),
+                     DstPortNumber = coalesce(DstPortNumber, host_parts[1], ipv4_parts[1],ipv6_parts[1])
+            | invoke _ASIM_ResolveDstFQDN('DstDomainPart')
+            | extend temp_is_MatchSrcHostname = DstHostname has_any (hostname_has_any), 
```

Comment:
This is post-filtering. Prefiltering requires checking if hostname_has_any is in RequestURL.